test/functional/logforwarding/message_format_test.go

```diff
@@ -0,0 +1,61 @@
+package logforwarding
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/test/functional"
+	"github.com/openshift/cluster-logging-operator/test/helpers"
+)
+
+var _ = Describe("[LogForwarding] Functional tests for message format", func() {
+
+	var (
+		framework *functional.FluentdFunctionalFramework
+	)
+
+	BeforeEach(func() {
+		framework = functional.NewFluentdFunctionalFramework()
+		functional.NewClusterLogForwarderBuilder(framework.Forwarder).
+			FromInput(logging.InputNameApplication).
+			ToFluentForwardOutput()
+		Expect(framework.Deploy()).To(BeNil())
+	})
+	AfterEach(func() {
+		framework.Cleanup()
+	})
+
+	It("should parse application log format correctly", func() {
+
+		// Define a template for test format (used for input, and expected output)
+		var testFormatTemplate = helpers.Log{
+			Timestamp: "2020-11-04T18:13:59.061892+00:00",
+			Message:   "Pre-Functional test message",
+			IndexName: "app-write",
+			Level:     "unknown",
+			MessageID: "*",
+			Docker:    &helpers.Docker{
+				ContainerID: "*"},
+			Kubernetes: &helpers.K8s{
+				ContainerName: "*",
+				PodName: "*",
+				NamespaceName:"*",
+				NamespaceID: "*",
+				OrphanedNamespace: "*"},
+		}
+
+		// Write log line as input to fluentd
+		applicationLogLine := fmt.Sprintf("%s stdout F %s $n", testFormatTemplate.Timestamp, testFormatTemplate.Message)
+		Expect(framework.WritesMessageToApplicationLogs(applicationLogLine,1)).To(BeNil())
+		// Read line from Log Forward output
+		raw, err := framework.ReadApplicationLogsFrom("fluentforward")
+		Expect(err).To(BeNil(), "Expected no errors reading the logs")
+		// Parse log line
+		logs, err := helpers.ParseLogs(raw)
+		Expect(err).To(BeNil(), "Expected no errors parsing the logs")
+		// Compare to expected template
+		outputTestLog := logs[0]
+		Expect(outputTestLog.CompareLog(testFormatTemplate)).To(BeTrue(), "Expected output to fit expected format")
```

Comment:
Consider converting this into a matcher like https://github.com/openshift/cluster-logging-operator/blob/master/test/matchers/envvar.go#L1

Reply:
Done